ow_power_system/src/power_system_node.cpp

```diff
@@ -17,71 +17,20 @@
-// discharge using a Monte Carlo predictor and an Unscented Kalman Filter.
-int simple_example() {
-    using namespace std::chrono;
-    // Read battery data from a file.
-    auto data = read_file("data_const_load.csv");
-
-    // Create a configuration from a file
-    ConfigMap config("example.cfg");
-
-    // Contruct a new prognoser using the prognoser factory. The prognoser
-    // will automatically construct an appropriate model, observer and predictor
-    // based on the values specified in the config.
-    auto prognoser = PrognoserFactory::instance().Create("ModelBasedPrognoser", config);
-
-    // For each line of data in the example file, run a single prediction step.
-    for (const auto& line : data) {
-        // Get a new prediction
-        auto prediction = prognoser->step(line);
-
-        // Get the event for battery EoD. The first line of data is used to initialize the observer,
-        // so the first prediction won't have any events.
-        if (prediction.getEvents().size() == 0) {
-            continue;
-        }
-        auto eod_event = prediction.getEvents().front();
-
-        // The time of event is a `UData` structure, which represents a data
-        // point while maintaining uncertainty. For the MonteCarlo predictor
-        // used by this example, the uncertainty is captured by storing the
-        // result of each particle used in the prediction.
-        UData eod_time = eod_event.getTOE();
-        if (eod_time.uncertainty() != UType::Samples) {
-            std::cerr << "Unexpected uncertainty type for EoD prediction" << std::endl;
-            return 1;
-        }
-
-        // For this example, we will print the median EoD.
-        auto samples = eod_time.getVec();
-        std::sort(samples.begin(), samples.end());
-        double eod_median = samples.at(samples.size() / 2);
-        auto now =  MessageClock::now();
-        auto now_s = duration_cast<std::chrono::seconds>(now.time_since_epoch());
-        std::cout << "Predicted median EoD: " << eod_median << " s (T- "
-                  << (eod_median-now_s.count())<< " s)" << std::endl;
-    }
-
-    return 0;
-}
-
 int main(int argc, char* argv[]) {
-  
+  using namespace std::chrono;
   ros::init(argc,argv,"power_system_node");
   ros::NodeHandle nh ("power_system_node");
 
-  //Construct our State of Charge (SOC) publisher (Volts)
+  //Construct our State of Charge (SOC) publisher (%)
   ros::Publisher SOC_pub = nh.advertise<std_msgs::Float64>("state_of_charge",1000);
   //Construct our Remaining Useful Life (RUL) publisher (Seconds)
   ros::Publisher RUL_pub = nh.advertise<std_msgs::Float64>("remaining_useful_life",1000);
   
   //Load power values csv
   string csv_path;
   auto csv_path_param_exist = nh.param("power_draw_csv_path", csv_path,
-    ros::package::getPath("ow_power_system") + "/data/onewatt.csv");
+    ros::package::getPath("ow_power_system") + "/data/data_const_load.csv");
```

Comment:
So the CSV file is hardcoded, and the one specified in launch file argument is ignored?

Reply:
I believe that the way it is written, the launch CSV file is hardcoded; however, if a csv_path_parameter exists in the launch file, that path overwrites the hardcoded path. The default (if not present) is to use the hardcoded path

Reply:
I believe that the way it is written, the launch CSV file is hardcoded; however, if a csv_path_parameter exists in the launch file, that path overwrites the hardcoded path. The default (if not present) is to use the hardcoded path

Reply:
I see.  This code would be clearer (especially to someone who doesn't know the 'param' interface) if you:

- replace 'auto' with 'bool'
- use a named constant for the default CSV

Reply:
That's fair. I will make this change

Reply:
Resolved